### DISCUSSION TITLE
Kubernetes custom resource name not passing

### DISCUSSION THREAD

**Discussion by BatBlob** [fabric8io/kubernetes-client#5471] [Q&A]
I'm trying to create a CRD context for an already deployed CRD, then fetch a list of it. However, fabric8 kubernetes client keeps throwing an exception where it cannot read the name of the resource.
```
CustomResourceDefinitionContext crd = new CustomResourceDefinitionContext.Builder()
                .withName("sparkapplications.sparkoperator.k8s.io")
                .withGroup("sparkoperator.k8s.io")
                .withScope("Namespaced")
                .withVersion("v1beta2")
                .withPlural("sparkapplications")
                .build();

GenericKubernetesResourceList listResponse = client.genericKubernetesResources(crd).inNamespace("default").list();
```

Error:
```
"exception":{"stacktrace":"io.fabric8.kubernetes.client.KubernetesClientException: Operation: [list]  for kind: [GenericKubernetesResource]  with name: [null]  in namespace: [default]  failed.
io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:159)
io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:426)
io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:392)
io.fabric8.kubernetes.client.dsl.internal.BaseOperation.list(BaseOperation.java:93)
```

My kubernetes client library version is 6.5.1 and kubernetes server/client version is v1.15.12.

The same code works with Kubernetes v1.24, but doesn't with kubernetes v1.15 (the one im using).

**Comment by rohanKanojia** [comment]
What error code are you getting? Your Kubernetes server version seems quite old.

**Reply by andreaTP** [reply]
I think the problem is with the `kind`: `Operation: [list]  for kind: [GenericKubernetesResource]`.

**Reply by rohanKanojia** [reply]
Umm, why do you think so? I remember list used to work with generic api.

**Reply by BatBlob** [reply]
Um, what do you mean by error code? The pod logs show this error, but it doesn't crash the pod, just skips the command.

> I think the problem is with the `kind`: `Operation: [list] for kind: [GenericKubernetesResource]`.

Also I think @andreaTP is right, because I tried adding `.withKind("SparkApplication")` in the custom resource definition context, but it still showed kind as GenericKubernetesResource in the error.

**Reply by BatBlob** [reply]
Another point, the same code works with Kubernetes v1.24, but doesn't with kubernetes v1.15 (the one im using).

**Reply by andreaTP** [reply]
I'm wondering how you are still on Kubernetes 1.15, as it's been out of maintenance for years now: https://endoflife.date/kubernetes

**Reply by rohanKanojia** [reply]
@BatBlob : [This Kubernetes guide](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#before-you-begin) seems to say Kubernetes 1.16 is minimum required version for using CustomResourceDefinitions
